app/screens/sso/sso_entra.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {StyleSheet, View} from 'react-native';
+
+import AuthError from './components/auth_error';
+import AuthRedirect from './components/auth_redirect';
+import AuthSuccess from './components/auth_success';
+
+interface SSOAuthenticationProps {
+    doEntraLogin: () => Promise<boolean>;
+    setLoginError: React.Dispatch<React.SetStateAction<string>>;
+    loginError: string;
+    theme: Theme;
+}
+
+const style = StyleSheet.create({
+    container: {
+        flex: 1,
+        paddingHorizontal: 24,
+    },
+});
+
+const SSOEntra = ({doEntraLogin, loginError, setLoginError, theme}: SSOAuthenticationProps) => {
+    const [loginSuccess, setLoginSuccess] = useState(false);
+
+    const init = useCallback(async () => {
+        setLoginSuccess(false);
+        setLoginError('');
+        const result = await doEntraLogin();
+        setLoginSuccess(result);
+    }, [doEntraLogin, setLoginError]);
+
+    useEffect(() => {
+        const timeout = setTimeout(() => {
+            init();
+        }, 1000);
```

Comment:
Any particular reason to wait one second? Should we put it in a constant?

Reply:
the reason is to give users "time" to read what is in the screen before this opens, same pattern as sso_authentication.tsx and ssp_authentication_with_external_browser.

can we make that a constant, possibly, important not really.